src/backend/utils/cache/typcache.c

```diff
@@ -1265,8 +1265,6 @@ set_record_type_typemod(int32 typmod, TupleDesc desc)
 void
 reset_record_type_cache(void)
 {
-	Assert(IsContQueryWorkerProcess() || IsContQueryAdhocProcess());
```

Comment:
The problem is that it gets called from `EXPLAIN`. That didn't used to be the case because this was called in `EndStreamScanBatch`, which doesn't happen anymore.